### PR TITLE
Add netcore50 configuration for System.IO

### PR DESCRIPTION
**Pull request by mellinoe**
As part of this, I've updated the buildtools version from 141 to 142. I needed to disable the inclusion of AssemblyInfoPartial as this library does not reference ResourceManager.

Also consolidated the source projects into a single file (deleted facade folder).

I did a quick pass of the source code to clean it and bring it in line with our guidelines. There are some additional cleanup items that we could tackle, like replacing some Contracts calls with Debug calls.

@weshaggard

### CODE REVIEW

File: src/System.IO/src/System.IO.builds [hunk 1]
```diff
@@ -3,9 +3,12 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
     <Project Include="System.IO.csproj" />
-    <Project Include="facade\System.IO.csproj">
+    <Project Include="System.IO.csproj">
       <TargetGroup>net46</TargetGroup>
     </Project>
+    <Project Include="System.IO.csproj">
+      <TargetGroup>netcore50aot</TargetGroup>
```
Comment (weshaggard) [inline src/System.IO/src/System.IO.builds]: Please add `<OSGroup>Windows_NT</OSGroup>`

File: src/System.IO/src/System.IO.csproj [hunk 2]
```diff
@@ -6,18 +6,68 @@
...
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net46_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net46_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50aot_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50aot_Release|AnyCPU'" />
+
+  <PropertyGroup Condition="'$(TargetGroup)' == 'net46'">
+    <ResourcesSourceOutputDirectory>None</ResourcesSourceOutputDirectory>
```
Comment (weshaggard) [inline src/System.IO/src/System.IO.csproj]: Should we add a proper way to disable this in BuildTools targets.

File: src/System.IO/src/System.IO.nuget.targets [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (weshaggard) [inline src/System.IO/src/System.IO.nuget.targets]: Lets not commit this file.

File: src/System.IO/src/System/IO/BinaryReader.cs [hunk 4]
```diff
@@ -0,0 +1,638 @@
...
+        }
+
+        public BinaryReader(Stream input, Encoding encoding) : this(input, encoding, false)
+        {
+        }
+
+        public BinaryReader(Stream input, Encoding encoding, bool leaveOpen)
+        {
+            if (input == null)
+            {
```
Comment (weshaggard) [inline src/System.IO/src/System/IO/BinaryReader.cs]: Can you make the bracing style consistent? for the argument checks?

File: src/System.IO/src/System/IO/BinaryReader.cs [hunk 5]
```diff
@@ -0,0 +1,638 @@
...
+                throw new ArgumentException(SR.Argument_StreamNotReadable);
+            Contract.EndContractBlock();
+            _stream = input;
+            _decoder = encoding.GetDecoder();
+            _maxCharsSize = encoding.GetMaxCharCount(MaxCharBytesSize);
+            int minBufferSize = encoding.GetMaxByteCount(1);  // max bytes per one char
+            if (minBufferSize < 16)
+                minBufferSize = 16;
+            _buffer = new byte[minBufferSize];
+            // m_charBuffer and m_charBytes will be left null.
```
Comment (weshaggard) [inline src/System.IO/src/System/IO/BinaryReader.cs]: Looks like a missed rename for m_charBuffer

File: src/System.IO/src/System/IO/BinaryReader.cs [hunk 6]
```diff
@@ -0,0 +1,638 @@
...
+        {
+            FillBuffer(8);
+            uint lo = (uint)(_buffer[0] | _buffer[1] << 8 |
+                             _buffer[2] << 16 | _buffer[3] << 24);
+            uint hi = (uint)(_buffer[4] | _buffer[5] << 8 |
+                             _buffer[6] << 16 | _buffer[7] << 24);
+            return ((ulong)hi) << 32 | lo;
+        }
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
```
Comment (weshaggard) [inline src/System.IO/src/System/IO/BinaryReader.cs]: You can remove the Security attributes.

File: src/System.IO/src/System/IO/Stream.cs [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+            }
+        }
+
+        [ComVisible(false)]
+        public Task WriteAsync(Byte[] buffer, int offset, int count)
+        {
+            return WriteAsync(buffer, offset, count, CancellationToken.None);
+        }
+
+        [ComVisible(false)]
```
Comment (weshaggard) [inline src/System.IO/src/System/IO/Stream.cs]: You can remove the ComVisible attributes.

File: src/System.IO/src/System.IO.nuget.targets (hunk 3 above)
Comment (mellinoe) [inline src/System.IO/src/System.IO.nuget.targets]: Agh, yes, this one always tries to sneak its way in. Will remove.

**Comment by mellinoe** [comment]
I went through and normalized all of the braces in the new code that I moved out, and replaced the usages of Contract.Assert and Requires with Debug.Assert. I've run the System.IO tests for .NET Native and they all pass.

**Comment by weshaggard** [comment]
LGTM

**Pull request merged**